provider/aws.go

```diff
@@ -460,6 +502,47 @@ func (p *AWSProvider) newChange(action string, endpoint *endpoint.Endpoint, reco
 		}
 	}
 
+	setIdentifier := endpoint.SetIdentifier
+	if setIdentifier != "" {
```

Comment:
can we also strip the code block out and put it into a new function and add some tests for that as well?

Reply:
Not sure if this is really necessary here, it just pulls some fields from the AWS representation of the record and puts it into the external-dns representation. There is no real logic to be tested here